front/src/modules/infra/useInfra.ts

```diff
@@ -8,8 +9,7 @@ import { useInfraID } from 'common/osrdContext';
  */
 export default function useInfra(infraID?: number) {
   const { data, isLoading, error } = osrdEditoastApi.endpoints.getInfraByInfraId.useQuery(
-    { infraId: infraID! },
-    { skip: isNil(infraID) }
+    !isNil(infraID) ? { infraId: infraID } : skipToken
```

Comment:
We can harmonize with the rest of the file where we use `infraID`
```suggestion
    infraID ? { infraId: infraID } : skipToken
```